src/Helper.php

```diff
@@ -123,30 +125,34 @@ function (string $className) {
             ) . 'Test';
         };
 
-        $specification = function (string $className) use ($testClassNameFrom) {
+        $classesWithoutTests = \array_filter($classyNames, function (string $className) use ($testClassNameFrom) {
             $reflection = new \ReflectionClass($className);
 
-            if ($reflection->isAbstract()
-                || $reflection->isInterface()
-                || $reflection->isTrait()
-                || $reflection->isSubclassOf(Framework\TestCase::class)
-            ) {
-                return true;
+            /**
+             * Construct is not concrete, does not need a test.
+             */
+            if ($reflection->isAbstract() || $reflection->isInterface() || $reflection->isTrait()) {
+                return false;
             }
 
-            $testClassName = $testClassNameFrom($className);
-
-            if (!\class_exists($testClassName)) {
+            /**
+             * Class is a test itself.
```

Comment:
Not entirely sure, this helps when test classes reside in the same namespace as source classes.